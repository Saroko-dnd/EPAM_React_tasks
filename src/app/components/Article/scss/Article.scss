@import "variables";

.news-article {
  box-shadow: 0 0 2rem 0 $shadow-color;

  h2 {
    font-size: 2rem;
  }

  a {
    font-size: 2rem;
  }
}
