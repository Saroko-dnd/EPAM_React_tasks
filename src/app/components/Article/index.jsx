import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

import { navigation } from '../../constants';

import './scss/Article.scss';

const Article = ({ title, articleId, url }) => {
    let detailsLink;
    let sourceLink;

    if (articleId) {
        detailsLink = (
            <NavLink to={navigation.newsDetails.destination + articleId} exact>
                read more
            </NavLink>
        );
    }

    if (url) {
        sourceLink = <a href={url}>read source</a>;
    }

    return (
        <article className="news-article d-flex align-items-center flex-column">
            <h2>{title}</h2>
            {detailsLink}
            {sourceLink}
        </article>
    );
};

Article.propTypes = {
    title: PropTypes.string.isRequired,
    articleId: PropTypes.string,
    url: PropTypes.string,
};

Article.defaultProps = {
    articleId: null,
    url: null,
};

export default Article;
