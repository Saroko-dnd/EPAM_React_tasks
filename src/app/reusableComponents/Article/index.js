import React from 'react';
// import { NavLink } from 'react-router-dom';

import './Article.scss';

const Article = props => (
    <article className="news-article d-flex align-items-center flex-column">
        <h2>{props.title}</h2>
        <a href={props.linkToDetails}>read more</a>
    </article>
);

/* <NavLink
className=""
to={props.linkToDetails}
exact
/> */

export default Article;
