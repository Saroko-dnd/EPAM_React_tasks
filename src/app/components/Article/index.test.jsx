import Article from '.';

describe('render an Article', () => {
    it('should render Article correctly', () => {
        const wrapper = shallow(<Article title="rrr" />);
        expect(wrapper).toMatchSnapshot();
    });
});
