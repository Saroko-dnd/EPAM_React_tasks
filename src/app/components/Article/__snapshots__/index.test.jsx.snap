// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render an Article should render Article correctly 1`] = `
<article
  className="card news-article"
>
  <span
    className="fa fa-newspaper-o"
  />
  <div
    className="card-body d-flex align-items-center justify-content-end flex-column"
  >
    <h2
      className=" card-title"
    >
      rrr
    </h2>
    <a
      className="card-link"
      href={null}
    >
      read source
    </a>
  </div>
</article>
`;
